### ISSUE TITLE
Internal Server Error with nextcloud 13

### ISSUE THREAD

**Issue by mayanksuman** [nextcloud/cms_pico#49]
### Steps to reproduce
1. Install nextcloud 13 and pico_cms app.
2. Turn on mod_rewrite in nextcloud server configuration
```
RewriteEngine On
RewriteRule /sites/(.*) https://server/nextcloud/index.php/apps/cms_pico/pico/$1 [QSA,L]
```
3. Create a website in pico and try to access it.

### Expected behaviour
Site should open without any error.

### Actual behaviour
Internal server error while accessing a site.

## Server configuration detail

**Operating system:** Linux 4.9.59-v7+ #1047 SMP Sun Oct 29 12:19:23 GMT 2017 armv7l

**Webserver:** Apache/2.4.25 (Raspbian) (apache2handler)

**Database:** mysql 10.1.23

**PHP version:** 7.0.27-0+deb9u1
Modules loaded: Core, date, libxml, openssl, pcre, zlib, filter, hash, Reflection, SPL, session, standard, apache2handler, mysqlnd, PDO, xml, apcu, apc, calendar, ctype, curl, dom, mbstring, fileinfo, ftp, gd, gettext, iconv, imagick, intl, json, exif, mcrypt, mysqli, pdo_mysql, Phar, posix, readline, shmop, SimpleXML, sockets, sysvmsg, sysvsem, sysvshm, tokenizer, wddx, xmlreader, xmlwriter, xsl, zip, Zend OPcache

**Nextcloud version:** 13.0.0 - 13.0.0.14

**Updated from an older Nextcloud/ownCloud or fresh install:** fresh install

**Where did you install Nextcloud from:** using tar file from nextcloud.com

<details><summary>Signing status</summary>

Array
</details>

<details><summary>List of activated apps</summary>

```
Enabled:
 - activity: 2.6.1
 - admin_notifications: 1.0.1
 - apporder: 0.4.1
 - bookmarks: 0.10.1
 - bruteforcesettings: 1.0.3
 - calendar: 1.6.0
 - checksum: 0.3.5
 - cms_pico: 0.9.7
 - comments: 1.3.0
 - contacts: 2.0.1
 - dashboard: 4.0.5
 - dav: 1.4.6
 - deck: 0.3.0
 - federatedfilesharing: 1.3.1
 - federation: 1.3.0
 - files: 1.8.0
 - files_accesscontrol: 1.3.0
 - files_automatedtagging: 1.3.0
 - files_downloadactivity: 1.2.0
 - files_markdown: 2.0.1
 - files_pdfviewer: 1.2.0
 - files_sharing: 1.5.0
 - files_texteditor: 2.5.1
 - files_trashbin: 1.3.0
 - files_versions: 1.6.0
 - files_videoplayer: 1.2.0
 - firstrunwizard: 2.2.1
 - gallery: 18.0.0
 - issuetemplate: 0.3.0
 - limit_login_to_ip: 1.0.2
 - logreader: 2.0.0
 - lookup_server_connector: 1.1.0
 - mail: 0.7.9
 - metadata: 0.6.0
 - news: 12.0.1
 - nextcloud_announcements: 1.2.0
 - notes: 2.3.2
 - notifications: 2.1.2
 - oauth2: 1.1.0
 - orcid: 0.9.1
 - password_policy: 1.3.0
 - provisioning_api: 1.3.0
 - ransomware_protection: 1.1.0
 - serverinfo: 1.3.0
 - sharebymail: 1.3.0
 - survey_client: 1.1.0
 - systemtags: 1.3.0
 - tasks: 0.9.6
 - theming: 1.4.1
 - twofactor_backupcodes: 1.2.3
 - unsplash: 1.0.5
 - updatenotification: 1.3.0
 - workflowengine: 1.3.0
Disabled:
 - admin_audit
 - encryption
 - files_external
 - user_external
 - user_ldap

```
</details>

<details><summary>Configuration (config/config.php)</summary>

```
{
    "instanceid": "***REMOVED SENSITIVE VALUE***",
    "passwordsalt": "***REMOVED SENSITIVE VALUE***",
    "secret": "***REMOVED SENSITIVE VALUE***",
    "trusted_domains": [
        "10.103.15.200",
        "server"
    ],
    "datadirectory": "***REMOVED SENSITIVE VALUE***",
    "overwrite.cli.url": "http:\/\/10.103.15.200\/nextcloud",
    "dbtype": "mysql",
    "version": "13.0.0.14",
    "dbname": "***REMOVED SENSITIVE VALUE***",
    "dbhost": "***REMOVED SENSITIVE VALUE***",
    "dbport": "",
    "dbtableprefix": "oc_",
    "mysql.utf8mb4": true,
    "dbuser": "***REMOVED SENSITIVE VALUE***",
    "dbpassword": "***REMOVED SENSITIVE VALUE***",
    "memcache.local": "\\OC\\Memcache\\APCu",
    "proxy": "http:\/\/172.16.2.30:8080",
    "installed": true,
    "loglevel": 0
}
```
</details>

**Are you using external storage, if yes which one:** local

**Are you using encryption:** no

**Are you using an external user-backend, if yes which one:** No

## Client configuration

**Browser:** Mozilla/5.0 (X11; Linux x86_64; rv:58.0) Gecko/20100101 Firefox/58.0

**Operating system:** Debian Testing

## Logs

<details><summary>Nextcloud log</summary>

```
{"reqId":"7QDKAPhcPsC18SQygqWI","level":3,"time":"2018-02-11T02:45:09+00:00","remoteAddr":"10.103.15.150","user":"user","app":"index","method":"GET","url":"\/nextcloud\/index.php\/apps\/cms_pico\/pico\/dddd","message":"Exception: {\"Exception\":\"ParseError\",\"Message\":\"syntax error, unexpected '?'\",\"Code\":0,\"Trace\":\"#0 \\\/data\\\/web_data\\\/nextcloud\\\/lib\\\/composer\\\/composer\\\/ClassLoader.php(321): Composer\\\\Autoload\\\\includeFile('\\\/data\\\/web_data\\\/...')\\n#1 [internal function]: Composer\\\\Autoload\\\\ClassLoader->loadClass('Symfony\\\\\\\\Compone...')\\n#2 \\\/data\\\/web_data\\\/nextcloud\\\/apps\\\/cms_pico\\\/vendor\\\/picocms\\\/pico\\\/lib\\\/Pico.php(785): spl_autoload_call('Symfony\\\\\\\\Compone...')\\n#3 \\\/data\\\/web_data\\\/nextcloud\\\/apps\\\/cms_pico\\\/lib\\\/Pico.php(81): Pico->parseFileMeta('---\\\\nTitle: Sub ...', Array)\\n#4 \\\/data\\\/web_data\\\/nextcloud\\\/apps\\\/cms_pico\\\/vendor\\\/picocms\\\/pico\\\/lib\\\/Pico.php(994): OCA\\\\CMSPico\\\\Pico->parseFileMeta('---\\\\nTitle: Sub ...', Array)\\n#5 \\\/data\\\/web_data\\\/nextcloud\\\/apps\\\/cms_pico\\\/vendor\\\/picocms\\\/pico\\\/lib\\\/Pico.php(340): Pico->readPages()\\n#6 \\\/data\\\/web_data\\\/nextcloud\\\/apps\\\/cms_pico\\\/lib\\\/Service\\\/PicoService.php(166): Pico->run()\\n#7 \\\/data\\\/web_data\\\/nextcloud\\\/apps\\\/cms_pico\\\/lib\\\/Service\\\/PicoService.php(109): OCA\\\\CMSPico\\\\Service\\\\PicoService->getContentFromPico(Object(OCA\\\\CMSPico\\\\Model\\\\Website))\\n#8 \\\/data\\\/web_data\\\/nextcloud\\\/apps\\\/cms_pico\\\/lib\\\/Service\\\/WebsitesService.php(249): OCA\\\\CMSPico\\\\Service\\\\PicoService->getContent(Object(OCA\\\\CMSPico\\\\Model\\\\Website))\\n#9 \\\/data\\\/web_data\\\/nextcloud\\\/apps\\\/cms_pico\\\/lib\\\/Controller\\\/PicoController.php(103): OCA\\\\CMSPico\\\\Service\\\\WebsitesService->getWebpageFromSite('dddd', 'user', '')\\n#10 \\\/data\\\/web_data\\\/nextcloud\\\/apps\\\/cms_pico\\\/lib\\\/Controller\\\/PicoController.php(88): OCA\\\\CMSPico\\\\Controller\\\\PicoController->getPage('dddd', '')\\n#11 [internal function]: OCA\\\\CMSPico\\\\Controller\\\\PicoController->getRoot('dddd')\\n#12 \\\/data\\\/web_data\\\/nextcloud\\\/lib\\\/private\\\/AppFramework\\\/Http\\\/Dispatcher.php(161): call_user_func_array(Array, Array)\\n#13 \\\/data\\\/web_data\\\/nextcloud\\\/lib\\\/private\\\/AppFramework\\\/Http\\\/Dispatcher.php(91): OC\\\\AppFramework\\\\Http\\\\Dispatcher->executeController(Object(OCA\\\\CMSPico\\\\Controller\\\\PicoController), 'getRoot')\\n#14 \\\/data\\\/web_data\\\/nextcloud\\\/lib\\\/private\\\/AppFramework\\\/App.php(115): OC\\\\AppFramework\\\\Http\\\\Dispatcher->dispatch(Object(OCA\\\\CMSPico\\\\Controller\\\\PicoController), 'getRoot')\\n#15 \\\/data\\\/web_data\\\/nextcloud\\\/lib\\\/private\\\/AppFramework\\\/Routing\\\/RouteActionHandler.php(47): OC\\\\AppFramework\\\\App::main('OCA\\\\\\\\CMSPico\\\\\\\\Con...', 'getRoot', Object(OC\\\\AppFramework\\\\DependencyInjection\\\\DIContainer), Array)\\n#16 [internal function]: OC\\\\AppFramework\\\\Routing\\\\RouteActionHandler->__invoke(Array)\\n#17 \\\/data\\\/web_data\\\/nextcloud\\\/lib\\\/private\\\/Route\\\/Router.php(297): call_user_func(Object(OC\\\\AppFramework\\\\Routing\\\\RouteActionHandler), Array)\\n#18 \\\/data\\\/web_data\\\/nextcloud\\\/lib\\\/base.php(998): OC\\\\Route\\\\Router->match('\\\/apps\\\/cms_pico\\\/...')\\n#19 \\\/data\\\/web_data\\\/nextcloud\\\/index.php(37): OC::handleRequest()\\n#20 {main}\",\"File\":\"\\\/data\\\/web_data\\\/nextcloud\\\/apps\\\/news\\\/vendor\\\/symfony\\\/yaml\\\/Parser.php\",\"Line\":510}","userAgent":"Mozilla\/5.0 (X11; Linux x86_64; rv:58.0) Gecko\/20100101 Firefox\/58.0","version":"13.0.0.14"}

```
</details>

**Comment by Ludovicis** [comment]
Same probleme with Mod_Proxy
Nextcloud version: 13.0.0 - 13.0.0.14
Pico CMS	0.9.6
Webserver: Apache/2.4.25 (Raspbian) (apache2handler)
Database: mysql 10.1.23
PHP version: 7.0.27-0+deb9u1
Same Nextcloud Log

**Comment by Ludovicis** [comment]
Same problem with Pico CMS 0.9.7

**Comment by piggz** [comment]
Im also getting an internal error.  In the nextcloud log I have


  | ParseError: syntax error, unexpected '?'/srv/piggz.co.uk/web/nextcloud11/lib/composer/composer/ClassLoader.php - line 321:  Composer\Autoload\includeFile('/srv/piggz.co.u...')[internal function] Composer\Autoload\ClassLoader->loadClass('Symfony\\Compone...')/srv/piggz.co.uk/web/nextcloud11/apps/cms_pico/vendor/picocms/pico/lib/Pico.php - line 785:  spl_autoload_call('Symfony\\Compone...')/srv/piggz.co.uk/web/nextcloud11/apps/cms_pico/lib/Pico.php - line 81:  Pico->parseFileMeta('---\nTitle  Welc...', Array)/srv/piggz.co.uk/web/nextcloud11/apps/cms_pico/vendor/picocms/pico/lib/Pico.php - line 321:  OCA\CMSPico\Pico->parseFileMeta('---\nTitle  Welc...', Array)/srv/piggz.co.uk/web/nextcloud11/apps/cms_pico/lib/Service/PicoService.php - line 166:  Pico->run()/srv/piggz.co.uk/web/nextcloud11/apps/cms_pico/lib/Service/PicoService.php - line 109:  OCA\CMSPico\Service\PicoService->getContentFromPico(Object(OCA\CMSPico\Model\Website))/srv/piggz.co.uk/web/nextcloud11/apps/cms_pico/lib/Service/WebsitesService.php - line 249:  OCA\CMSPico\Service\PicoService->getContent(Object(OCA\CMSPico\Model\Website))/srv/piggz.co.uk/web/nextcloud11/apps/cms_pico/lib/Controller/PicoController.php - line 103:  OCA\CMSPico\Service\WebsitesService->getWebpageFromSite('pgz', NULL, '')/srv/piggz.co.uk/web/nextcloud11/apps/cms_pico/lib/Controller/PicoController.php - line 88:  OCA\CMSPico\Controller\PicoController->getPage('pgz', '')[internal function] OCA\CMSPico\Controller\PicoController->getRoot('pgz')/srv/piggz.co.uk/web/nextcloud11/lib/private/AppFramework/Http/Dispatcher.php - line 161:  call_user_func_array(Array, Array)/srv/piggz.co.uk/web/nextcloud11/lib/private/AppFramework/Http/Dispatcher.php - line 91:  OC\AppFramework\Http\Dispatcher->executeController(Object(OCA\CMSPico\Controller\PicoController), 'getRoot')/srv/piggz.co.uk/web/nextcloud11/lib/private/AppFramework/App.php - line 115:  OC\AppFramework\Http\Dispatcher->dispatch(Object(OCA\CMSPico\Controller\PicoController), 'getRoot')/srv/piggz.co.uk/web/nextcloud11/lib/private/AppFramework/Routing/RouteActionHandler.php - line 47:  OC\AppFramework\App  main('OCA\\CMSPico\\Con...', 'getRoot', Object(OC\AppFramework\DependencyInjection\DIContainer), Array)[internal function] OC\AppFramework\Routing\RouteActionHandler->__invoke(Array)/srv/piggz.co.uk/web/nextcloud11/lib/private/Route/Router.php - line 297:  call_user_func(Object(OC\AppFramework\Routing\RouteActionHandler), Array)/srv/piggz.co.uk/web/nextcloud11/lib/base.php - line 998:  OC\Route\Router->match('/apps/cms_pico/...')/srv/piggz.co.uk/web/nextcloud11/index.php - line 37:  OC  handleRequest(){main}
-- | --

**Comment by diplodata** [comment]
This might work https://github.com/nextcloud/cms_pico/issues/55

**Comment by Windyo** [comment]
same issue. Checked #55 but seems to be ubuntu-specific, i run debian.

**Comment by piggz** [comment]
Upgrading PHP to 7.2 fixed it for me

**Comment by gerroon** [comment]
Sorry but suggestion to move to php 7.2 is a bold move, not all systems are ready. Is there a fix for 7.1?

**Comment by Windyo** [comment]
OK so it's not actually php.

if anyone's still reading this, most the times this is because you're using NextCloudPi and you're telling apache2 to use mod_proxy without having a proxy module installed.

Do `sudo a2enmod proxy_http`

and _voilà_.
